Added support for metalink repo defintions

schaefi opened the pull request:
Repository source paths that points to a metalink file which
needs to be downloaded and handled by the configured package
manager requires a different repository configuration. Thus
we now support a new attribute metalink which allows to flag
the repository definition as a metalink source. Currently
the dnf repository class is the only one that makes use of
the information. This change is required to support
Fedora >= 30 images. I also expect this concept to be used
by other distributions in the future

Conan-Kudo commented:
@schaefi It'd be good if you also added `mirrorlist` since that's used by CentOS too...

Conan-Kudo commented:
Note that `baseurl`, `metalink`, or `mirrorlist` are more or less mutually exclusive.

schaefi commented:
> @schaefi It'd be good if you also added `mirrorlist` since that's used by CentOS too...

Ok, I've changed the code to provide the attribute ```sourcetype``` which allows one of the values ```baseurl```, ```metalink``` or ```mirrorlist```
